homeassistant/components/homewizard_climate/__init__.py

```diff
@@ -0,0 +1,61 @@
+"""The Homewizard Climate integration."""
+from __future__ import annotations
+
+import logging
+
+from homewizard_climate_websocket.api.api import HomeWizardClimateApi
+from homewizard_climate_websocket.model.climate_device import HomeWizardClimateDevice
+from homewizard_climate_websocket.ws.hw_websocket import HomeWizardClimateWebSocket
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, PASSWORD, USERNAME
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Homewizard Climate from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    api: HomeWizardClimateApi = HomeWizardClimateApi(
+        entry.data.get(USERNAME), entry.data.get(PASSWORD)
+    )
```

Comment:
Can this catch password (re)authentication?

Maybe raise `InvalidAuthError` when auth failed? Otherwise users have to remove and add the integration.

Reply:
- The constructor of this class does not perform a login automatically, it needs to be called directly with `api.login()`
- An initial login is attempted in the [config flow](https://github.com/mepla/home-assistant-core/blob/homewizard_climate/homeassistant/components/homewizard_climate/config_flow.py#L33-L39) before reaching this point, and a proper `InvalidAuthError` raised there.
- re-auth is a handled within the library itself (upon expired tokens).

Reply:
Aha, I misunderstood re-auth with "refreshing the token", but I guess you meant something like [this](https://developers.home-assistant.io/docs/config_entries_config_flow_handler?_highlight=reauth#reauthentication). I will look into that now.

Reply:
Yes 👍

Reply:
Reauth flow is implemented now: https://github.com/home-assistant/core/pull/84105/commits/7fa778c0749b061fc741ec3fde6de2455bc78b84